### PR TITLE
Get-DbaLinkedServer

### PR DESCRIPTION
**Pull request by Staggerlee011**
v1

### CODE REVIEW

File: functions/Get-DbaLinkedServer.ps1 [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+                        RemoteServer = $ls.DataSource
+                        ProductName = $ls.ProductName 
+                        Impersonate = $ls.LinkedServerLogins.Impersonate
+                        RemoteUser = $ls.LinkedServerLogins.remoteuser
+                        Rpc = $ls.Rpc
+                        RpcOut = $ls.RpcOut
+                        LinkedServer = $ls
+                        }
+     
+                    Select-DefaultField -InputObject $output -Property ComputerName, SqlInstance, LinkedServerName, RemoteServer, ProductName, Impersonate, RemoteUser, Rpc, RpcOut
```
Comment (SirCaptainMitch) [inline functions/Get-DbaLinkedServer.ps1]: Not sure if I am missing something on this line, but this errors out. 

`Select-DefaultField : The term 'Select-DefaultField' is not recognized as the name of a cmdlet, function, script file, or 
operable program.`

Comment (wsmelton) [inline functions/Get-DbaLinkedServer.ps1]: @SirCaptainMitch that is an internal function...if you are testing it just make sure you have that command in your dbatools\internals folder. It was just added about 6 days ago.

File: functions/Get-DbaLinkedServer.ps1 [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	        }
+            if($server -contains "\")
+            {
+                $computername = ($server.Name).Split("\")[0]
+                $inst =  ($server.Name).Split("\")[1]
+            }
+            else
+            {
+                $computername = $server.Name
+                $inst =  ""
```
Comment (wsmelton) [inline functions/Get-DbaLinkedServer.ps1]: Best to have this as `$inst = $null`

File: functions/Get-DbaLinkedServer.ps1 [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        {
+            try
+	        {
+	            Write-Verbose "Connecting to $Instance"
+                $server = Connect-SqlServer -SqlServer $Instance -SqlCredential $sqlcredential
+	        }
+	        catch
+	        {
+	            Write-Warning "Failed to connect to: $srv"
+                break
```
Comment (wsmelton) [inline functions/Get-DbaLinkedServer.ps1]: This should be `continue`. Using `break` will cause it to go out of the foreach loop if one of many servers is passed in and you hit one that throws an exception.

File: functions/Get-DbaLinkedServer.ps1 (hunk 2 above)
Comment (wsmelton) [inline functions/Get-DbaLinkedServer.ps1]: As well you can actually  just remove the else all together. 

Also you don't need to use the extra parenthesizes to split it, and just another way to do it:

```powershell
if ($server.Contains("\") {
   $computername = $server.Name.Split("\")[0]
   $inst = $server.Name.Split("\")[1]
}
```

The above will ensure you still get the computer name, and if you deal with a default instance, the `$inst` will be `null` and not an empty string.

File: functions/Get-DbaLinkedServer.ps1 [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+	
+.LINK
+https://dbatools.io/Get-DbaLinkedServer
+
+.EXAMPLE
+Get-DbaLinkedServer -SqlServer DEV01
+
+Returns all Linked Servers for the SQL instance DEV01
```
Comment (wsmelton) [inline functions/Get-DbaLinkedServer.ps1]: This should read

> Returns all Linked Servers for the SQL Server instance DEV01

File: functions/Get-DbaLinkedServer.ps1 [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+.DESCRIPTION
+Retrieves information about each linked server on the instance
+
+.PARAMETER SqlServer
+The SQL Server that you're connecting to.
+
+.PARAMETER Credential
+Credential object used to connect to the SQL Server as a different user
+
+.PARAMETER Databases
```
Comment (wsmelton) [inline functions/Get-DbaLinkedServer.ps1]: This parameter should be removed since it is not being used.

File: functions/Get-DbaLinkedServer.ps1 [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+.SYNOPSIS
+Gets all linked servers and summary of information from the sql servers listed
+
+.DESCRIPTION
+Retrieves information about each linked server on the instance
+
+.PARAMETER SqlServer
+The SQL Server that you're connecting to.
+
+.PARAMETER Credential
```
Comment (wsmelton) [inline functions/Get-DbaLinkedServer.ps1]: Update the parameter name to `$SqlCredential` to match the parameters for the function.

**Comment by ctrlbold** [comment]
These get commands should return full on SMO objects ala Get-DbaDatabase (which I haven't finished ;) I will approve and fix

**Pull request merged**